mem/vm/mmu/mmu.go

```diff
@@ -346,6 +357,53 @@ func unique(intSlice []uint64) []uint64 {
 	return list
 }
 
+func (m *middleware) createDefaultPage(pid vm.PID, vAddr uint64, deviceID uint64) vm.Page {
+	// Align virtual address to page boundary
+	alignedVAddr := (vAddr >> m.log2PageSize) << m.log2PageSize
+	pageSize := uint64(1) << m.log2PageSize
+	
+	// Allocate the smallest available physical page
+	pAddr := m.allocatePhysicalPage()
+	
+	return vm.Page{
+		PID:         pid,
+		VAddr:       alignedVAddr,
+		PAddr:       pAddr,
+		PageSize:    pageSize,
+		Valid:       true,
+		DeviceID:    deviceID,
+		Unified:     true,
+		IsMigrating: false,
+		IsPinned:    false,
+	}
+}
+
+// allocatePhysicalPage finds and allocates the smallest available physical page.
+// It returns the physical address of the allocated page.
+func (m *middleware) allocatePhysicalPage() uint64 {
+	pageSize := uint64(1) << m.log2PageSize
+	
+	// Find the smallest available physical page
+	for {
+		// Align to page boundary
+		candidatePage := (m.nextPhysicalPage >> m.log2PageSize) << m.log2PageSize
+		
+		// Check if this page is already in use
+		if !m.usedPhysicalPages[candidatePage] {
+			// Mark this page as used
+			m.usedPhysicalPages[candidatePage] = true
+			
+			// Update next physical page for future allocations
+			m.nextPhysicalPage = candidatePage + pageSize
+			
+			return candidatePage
```

Comment:
We also need to check the page table to ensure the page is not allocated in other places.